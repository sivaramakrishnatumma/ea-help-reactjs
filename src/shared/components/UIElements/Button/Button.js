import React from "react";

const Button = props => {
  return (
    <button
      className={"btn " + props.className}
      onClick={props.onClick}
      type={props.type}
    >
      {props.children}
    </button>
  );
};

export default Button;