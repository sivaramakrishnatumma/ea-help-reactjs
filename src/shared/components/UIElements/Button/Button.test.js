import React from "react";
import Button from "./Button";
import { mount } from "enzyme";

describe("Button component test case", () => {
  it("should test button component with default props", () => {
    const props = {
      className: "btn-primary",
      type: "submit",
      onClick: jest.fn()
    };
    const wrapper = mount(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
