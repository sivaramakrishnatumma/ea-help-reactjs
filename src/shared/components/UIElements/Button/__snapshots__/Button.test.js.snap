// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test case should test button component with default props 1`] = `
<Button
  className="btn-primary"
  onClick={[MockFunction]}
  type="submit"
>
  <button
    className="btn btn-primary"
    onClick={[MockFunction]}
    type="submit"
  />
</Button>
`;
